src/state.py

```diff
@@ -155,6 +156,40 @@ def _get_agent_meta_map_from_relation(
     return {unit.name: AgentMeta.from_agent_relation(relation.data[unit]) for unit in remote_units}
 
 
+class ProxyConfig(BaseModel):
+    """Configuration for accessing Jenkins through proxy.
+
+    Attributes:
+        http_proxy: The http proxy URL.
+        https_proxy: The https proxy URL.
+        no_proxy: Comma separated list of hostnames to bypass proxy.
+    """
+
+    http_proxy: typing.Optional[HttpUrl]
+    https_proxy: typing.Optional[HttpUrl]
+    no_proxy: typing.Optional[str]
+
+    @classmethod
+    def from_charm_env(cls, env: typing.Mapping[str, str]) -> typing.Optional["ProxyConfig"]:
+        """Instantiate ProxyConfig from juju charm environment.
+
+        Args:
+            env: The charm environment variable.
+
+        Returns:
+            ProxyConfig if proxy configuration is provided, None otherwise.
+        """
+        http_proxy = env.get("HTTP_PROXY")
+        https_proxy = env.get("HTTPS_PROXY")
+        no_proxy = env.get("NO_PROXY")
```

Comment:
Should we listen to `JUJU_CHARM_HTTP_PROXY`, `JUJU_CHARM_HTTPS_PROXY` and `JUJU_CHARM_NO_PROXY` instead?

Reply:
Yeh I think these environment variables are not quite right

Reply:
Hm interesting... I've tried printing the `env` and this is what I get (irrelevant envs removed)
```
{
  "JUJU_UNIT_NAME": "jenkins-k8s/0",
  "HTTPS_PROXY": "http://192.168.0.196:3128",
  "no_proxy": "127.0.0.1,::1,localhost",
  "JUJU_VERSION": "2.9.43",
  "JUJU_CHARM_HTTP_PROXY": "",
  "NO_PROXY": "127.0.0.1,::1,localhost",
  "JUJU_CHARM_HTTPS_PROXY": "",
  "JUJU_MODEL_NAME": "stg-jenkins-k8s",
  "https_proxy": "http://192.168.0.196:3128",
  "http_proxy": "http://192.168.0.196:3128",
  "JUJU_CHARM_DIR": "/var/lib/juju/agents/unit-jenkins-k8s-0/charm",
  "JUJU_CHARM_NO_PROXY": "127.0.0.1,localhost,::1",
  "HTTP_PROXY": "http://192.168.0.196:3128",
  "PYTHONPATH": "lib:venv",
}
```
I can see that there is `http_proxy` and `https_proxy` but none of `JUJU_CHARM_HTTP_PROXY`, `JUJU_CHARM_HTTPS_PROXY`, `JUJU_CHARM_NO_PROXY`. Would this be due to my juju version?

Reply:
https://chat.charmhub.io/charmhub/pl/xjjptuhuxjgcffr5pg3rxkhtxw

Asked for clarification of which environment variable should be used on public mattermost!

Reply:
Changed, thank you!